src/libs/actions/Transaction.ts

```diff
@@ -127,9 +128,10 @@ function removeWaypoint(transaction: Transaction, currentIndex: string, isDraft:
     // to remove nested keys while also preserving other object keys
     // Doing a deep clone of the transaction to avoid mutating the original object and running into a cache issue when using Onyx.set
     let newTransaction: Transaction = {
-        ...transaction,
+        // eslint-disable-next-line @typescript-eslint/non-nullable-type-assertion-style
+        ...(transaction as Transaction),
```

Comment:
This looks bad 😅 We need a better way to handle null transaction, maybe an early return?

Reply:
@kubabutkiewicz bump